src/ring_theory/nilpotent.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2021 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import data.nat.choose.sum
+
+/-!
+# Nilpotent elements
+
+## Main definitions
+
+  * `is_nilpotent`
+  * `is_nilpotent_neg_iff`
+  * `commute.is_nilpotent_add`
+  * `commute.is_nilpotent_mul_left`
+  * `commute.is_nilpotent_mul_right`
+  * `commute.is_nilpotent_sub`
+
+## Tags
+
+nilpotent, subhas_mul, subhas_add
```

Comment:
`subhas_mul`?

Reply:
Sorry, that's garbage I forgot to remove.

Reply:
(it was supposed to be a reminder to myself to make the point that the nilpotent elements of a commutative ring are naturally an additive subgroup closed under multiplication but not containing the multiplicative unit)